daemon/container.go

```diff
@@ -124,6 +124,29 @@ func (daemon *Daemon) newContainer(name string, platform ocispec.Platform, confi
 		return nil, err
 	}
 
+	// Check if any mount is of type apisocket and set DOCKER_HOST environment variable
+	for i, mnt := range hostConfig.Mounts {
+		if mnt.Type == mount.TypeAPISocket {
+			// TODO(ndeloof) engineSocket should create a dedicated socket for id
```

Comment:
Need to investigate how to register an additional listener (if feasible? Seems this would create a server -> daemon -> sever cycle)